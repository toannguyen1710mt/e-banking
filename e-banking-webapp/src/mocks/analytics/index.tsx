export const ANALYTICS_DATA = [
  {
    title: 'Total Income',
    subtitle: 'Last 30 days income',
    isPositive: true,
    amount: 670000,
    percentageChange: 8,
  },
  {
    title: 'Total Expense',
    subtitle: 'Last 30 days income',
    isPositive: true,
    amount: 270000,
    percentageChange: 8,
  },
  {
    title: 'Total Investment',
    subtitle: 'Last 30 days income',
    isPositive: true,
    amount: 170000,
    percentageChange: 8,
  },
];
