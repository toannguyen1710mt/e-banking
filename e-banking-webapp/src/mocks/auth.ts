// Libs
import { Session } from 'next-auth';

// Interfaces
import { IUser } from '@/interfaces';

export const MOCK_DATA_USER: IUser = {
  accounts: [
    {
      id: 114,
      documentId: 'e5bavd05yrng3g850yazqj0q',
      name: 'OCB',
      accountNumber: '123456789208',
      balance: 3441800,
      type: 'Main',
      currency: '$',
    },
    {
      id: 115,
      documentId: 'kvelrmbr5yan4gyhz4wnoio3',
      name: 'ACB',
      accountNumber: '999332289888',
      balance: 84438800,
      type: 'Checkings',
      currency: '$',
    },
    {
      id: 116,
      documentId: 'nios4sn0bgw5u28nelo8fmao',
      name: 'VID',
      accountNumber: '106868929357',
      balance: 9235800,
      type: 'Savings',
      currency: '$',
    },
  ],
  avatar: 'https://i.ibb.co/SKHPQYq/avatar-default.webp',
  country: 'UK',
  phone: '123456789000',
  postal: '226',
  token:
    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ODIsImlhdCI6MTczNzA4NzMwNiwiZXhwIjoxNzM5Njc5MzA2fQ.6VPWVxS169pVk-4xBcC9TOj2e1DS10b_l3t9Grs5pU0',
  id: 82,
  documentId: 'ldrhsspx98g1ijlabnrclht8',
  username: 'phong',
  email: 'phong+1@gmail.com',
  provider: 'local',
  confirmed: true,
  blocked: false,
  announcements: true,
  updates: true,
  feedbacksAndSurvey: true,
  events: true,
  generalNotification: true,
  promotions: true,
  eventsNearMe: true,
  createdAt: '2025-01-15T04:54:45.295Z',
  updatedAt: '2025-01-15T04:54:45.623Z',
  publishedAt: '2025-01-15T04:54:45.296Z',
};

export const MOCK_DATA_USER_EMPTY_PROPS: IUser = {
  accounts: [
    {
      documentId: '11',
      accountNumber: '',
      balance: 0,
      currency: '$',
      id: 36,
      name: '',
      type: '',
    },
    {
      documentId: '12',
      accountNumber: '',
      balance: 0,
      currency: '$',
      id: 37,
      name: '',
      type: '',
    },
    {
      documentId: '12',
      accountNumber: '',
      balance: 0,
      currency: '$',
      id: 38,
      name: '',
      type: '',
    },
  ],
  avatar: 'https://i.ibb.co/SKHPQYq/avatar-default.webp',
  country: 'UK',
  email: 'thong+1@gmail.com',
  id: 43,
  phone: '123456789000',
  postal: '226',
  token: '123456',
  username: 'thong',
  documentId: '',
  provider: '',
  confirmed: false,
  blocked: false,
  createdAt: '',
  updatedAt: '',
  publishedAt: '',
};

export const MOCK_SESSION_DATA: Session = {
  user: {
    email: 'phong+1@gmail.com',
    id: 82,
    documentId: 'ldrhsspx98g1ijlabnrclht8',
    username: 'phong',
    provider: 'local',
    confirmed: true,
    blocked: false,
    postal: '344',
    phone: '324345346345',
    country: 'United States',
    avatar: 'https://i.ibb.co/SKHPQYq/avatar-default.webp',
    announcements: true,
    updates: true,
    feedbacksAndSurvey: true,
    events: true,
    generalNotification: true,
    promotions: true,
    eventsNearMe: true,
    createdAt: '2025-01-15T04:54:45.295Z',
    updatedAt: '2025-01-15T04:54:45.623Z',
    publishedAt: '2025-01-15T04:54:45.296Z',
    token:
      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ODIsImlhdCI6MTczNzA4NzMwNiwiZXhwIjoxNzM5Njc5MzA2fQ.6VPWVxS169pVk-4xBcC9TOj2e1DS10b_l3t9Grs5pU0',
  },
  expires: '2025-01-17T04:24:13.935Z',
};
