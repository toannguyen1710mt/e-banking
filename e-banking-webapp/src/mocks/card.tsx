export const ANALYTICS_CARD = [
  {
    title: 'Total Income',
    isPositive: true,
    subtitle: 'Last 30 days income',
  },
  {
    title: 'Total Expense',
    isPositive: true,
    subtitle: 'Last 30 days expenditure',
  },
  {
    title: 'Total Investment',
    subtitle: 'Last 30 days investment',
  },
];

export const MOCK_LIST_CARD = [
  {
    id: 1,
    documentId: 'zj103n8cbv857otlqpt3ks2h',
    holderName: 'donal trump',
    cardNumber: '123412341234',
    expireAt: '2026-05',
    ccv: '336',
    fullName: null,
    createdAt: '2024-12-18T02:37:12.625Z',
    updatedAt: '2024-12-18T02:37:12.625Z',
    publishedAt: '2024-12-18T02:37:12.623Z',
  },
  {
    id: 2,
    documentId: 'ny4mxu6br8v900c3an9p5u0f',
    holderName: 'Beck Kham',
    cardNumber: '106868929357',
    expireAt: '2028-03',
    ccv: '339',
    fullName: null,
    createdAt: '2024-12-18T04:27:04.764Z',
    updatedAt: '2024-12-18T04:27:04.764Z',
    publishedAt: '2024-12-18T04:27:04.762Z',
  },
  {
    id: 3,
    documentId: 'ljq5kv0f327w5sor2czgs07t',
    holderName: 'Hoang Tran',
    cardNumber: '123412341234',
    expireAt: '2025-07',
    ccv: '333',
    fullName: null,
    createdAt: '2024-12-18T18:24:58.801Z',
    updatedAt: '2024-12-18T18:24:58.801Z',
    publishedAt: '2024-12-18T18:24:58.799Z',
  },
  {
    id: 4,
    documentId: 's0bz93jb4tc5w7wc820o33c2',
    holderName: 'Test Holder Name',
    cardNumber: '123456789876',
    expireAt: '2026-02',
    ccv: '123',
    fullName: null,
    createdAt: '2024-12-19T01:39:29.501Z',
    updatedAt: '2024-12-19T01:39:29.501Z',
    publishedAt: '2024-12-19T01:39:29.498Z',
  },
];
