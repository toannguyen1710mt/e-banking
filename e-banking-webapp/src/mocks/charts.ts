export const MOCK_SPENDING_STATISTIC_CHART_DATA = {
  series: [
    {
      name: 'Data 1',
      data: [10, 41, 35, 51, 49, 91, 133],
    },
    {
      name: 'Data 2',
      data: [28, 69, 73, 102, 123, 142, 153],
    },
  ],
  categories: ['Mon', 'Tue', 'Wed', 'Thur', 'Fri', 'Sat', 'Sun'],
};

export const MOCK_BALANCE_STATISTICS_CHART_DATA = {
  series: [
    {
      name: 'Data 1',
      data: [98, 124, 166, 350, 40, 289, 96],
    },
    {
      name: 'Data 2',
      data: [80, 132, 169, 270, 49, 174, 90],
    },
  ],
  categories: ['Mon', 'Tue', 'Wed', 'Thur', 'Fri', 'Sat', 'Sun'],
};

export const MOCK_SERIES_EXPENSE_ANALYSIS = [22, 45, 35, 40];

export const MASTERCARD_CHART_MOCK = [65, 20, 15];
