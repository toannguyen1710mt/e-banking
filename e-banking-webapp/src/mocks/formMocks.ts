import { AccountType, GlobalType } from '@/interfaces';
import { useForm } from 'react-hook-form';

export const useMockFormGlobal = () => {
  return useForm({
    defaultValues: {
      globalTransfer: {
        fromAccountType: '' as AccountType,
        fromCountryType: '' as GlobalType,
        recipientAccount: '',
        amount: '',
      },
      fromAccountId: '',
      fromCardName: '',
      fromAccountNumber: '',
      fromAccountBalance: 0,
      recipientName: '',
    },
  });
};

export const useMockFormInternal = () => {
  return useForm({
    defaultValues: {
      fromAccountId: '',
      fromCardName: '',
      fromAccountNumber: '',
      fromAccountBalance: 0,
      toAccountId: '',
      internalTransfer: {
        amount: '',
        fromAccountType: '' as AccountType,
        toAccountType: '' as AccountType,
      },
      toCardName: '',
      toAccountNumber: '',
      toAccountBalance: 0,
    },
  });
};
