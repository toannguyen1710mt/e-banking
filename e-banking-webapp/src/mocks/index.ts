export * from './options';
export * from './card';
export * from './charts';
export * from './transaction';
export * from './table';
export * from './transferConfirmData';
export * from './target';
export * from './analytics';
export * from './auth';
export * from './ServiceCard';
export * from './formMocks';
