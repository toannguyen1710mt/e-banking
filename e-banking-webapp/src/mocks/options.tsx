import Link from 'next/link';

// Constants
import { ROUTES } from '@/constants';

// Components
import { SettingIcon, SignOutIcon } from '@/components';

export const MOCK_OPTIONS = [
  {
    key: 'Kenya',
    label: 'Kenya',
  },
  {
    key: 'USA',
    label: 'USA',
  },
];

export const MOCK_CUSTOM_OPTIONS = [
  {
    key: 'Pheroxios Yehudi',
    label: '',
    customOptionElement: (
      <div className='flex flex-col gap-2'>
        <span className='text-xs font-semibold'>Pheroxios Yehudi</span>
        <span className='text-3xs font-semibold text-transparentBlack'>
          Pheroxios@yehudi.com
        </span>
      </div>
    ),
  },
  {
    key: 'Settings',
    label: 'Settings',
    startContent: <SettingIcon />,
    customOptionElement: <Link href={ROUTES.GENERAL}>Settings</Link>,
  },
  {
    key: 'Sign Out',
    label: 'Sign Out',
    startContent: <SignOutIcon />,
  },
];
