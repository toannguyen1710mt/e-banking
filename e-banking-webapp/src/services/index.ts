export * from './http-client';
export * from './account';
export * from './transaction';
export * from './users';
