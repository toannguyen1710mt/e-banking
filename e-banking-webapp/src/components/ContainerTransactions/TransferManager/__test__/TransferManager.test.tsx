// Libs
import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';

// API
import { getTransactionsByUserId } from '@/services';

// Mock
import { MOCK_SESSION_DATA } from '@/mocks';

// Components
import { TransferManager } from '../index';

beforeAll(() => {
  global.IntersectionObserver = jest.fn().mockImplementation((_callback) => ({
    observe: jest.fn(),
    unobserve: jest.fn(),
    disconnect: jest.fn(),
  }));
});

afterAll(() => {
  jest.restoreAllMocks();
});

jest.mock('@/services', () => ({
  getTransactionsByUserId: jest.fn(),
}));

const MOCK_TRANSACTIONS_RESPONSE = {
  meta: {
    pagination: {
      total: 5,
    },
  },
  data: [],
};

describe('TransferManager component', () => {
  beforeEach(() => {
    (getTransactionsByUserId as jest.Mock).mockResolvedValue(
      MOCK_TRANSACTIONS_RESPONSE,
    );
  });

  test('should match snapshot', async () => {
    const { container } = render(
      <TransferManager
        session={MOCK_SESSION_DATA}
        totalTransferReceived={5}
        totalTransferSent={5}
      />,
    );

    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });

  test('should call getTransactionsByUserId with correct parameters', async () => {
    render(
      <TransferManager
        session={MOCK_SESSION_DATA}
        totalTransferReceived={5}
        totalTransferSent={5}
      />,
    );

    await waitFor(() => {
      expect(getTransactionsByUserId).toHaveBeenCalledWith(
        MOCK_SESSION_DATA.user.id,
        expect.any(Object),
      );
    });
  });

  test('should display transactions received and sent', async () => {
    render(
      <TransferManager
        session={MOCK_SESSION_DATA}
        totalTransferReceived={5}
        totalTransferSent={5}
      />,
    );

    await waitFor(() => {
      expect(screen.getByText('Transfer Sent')).toBeInTheDocument();
    });
  });

  test('should fetch more transactions when scrolled to bottom', async () => {
    render(
      <TransferManager
        session={MOCK_SESSION_DATA}
        totalTransferReceived={5}
        totalTransferSent={5}
      />,
    );

    await waitFor(() => {
      expect(getTransactionsByUserId).toHaveBeenCalledTimes(4);
    });

    const observerCallback = (global.IntersectionObserver as jest.Mock).mock
      .calls[0][0];
    observerCallback([{ isIntersecting: true }]);

    await waitFor(() => {
      expect(getTransactionsByUserId).toHaveBeenCalledTimes(5);
    });
  });

  test('should switch transfer type and fetch transactions', async () => {
    render(
      <TransferManager
        session={MOCK_SESSION_DATA}
        totalTransferReceived={5}
        totalTransferSent={5}
      />,
    );

    await waitFor(() => {
      expect(getTransactionsByUserId).toHaveBeenCalledWith(
        MOCK_SESSION_DATA.user.id,
        expect.objectContaining({
          filters: { toAccountType: { $notNull: undefined } },
        }),
      );
    });

    screen.getByText('Transfer Sent').click();

    await waitFor(() => {
      expect(getTransactionsByUserId).toHaveBeenCalledWith(
        MOCK_SESSION_DATA.user.id,
        expect.objectContaining({
          filters: { toAccountType: { $null: undefined } },
        }),
      );
    });
  });

  test('should append new transactions to the existing list', async () => {
    const initialTransactions = [{ id: 1 }];
    const newTransactions = [{ id: 2 }, { id: 3 }];

    (getTransactionsByUserId as jest.Mock)
      .mockResolvedValueOnce({
        ...MOCK_TRANSACTIONS_RESPONSE,
        data: initialTransactions,
      })
      .mockResolvedValueOnce({
        ...MOCK_TRANSACTIONS_RESPONSE,
        data: newTransactions,
      });

    render(
      <TransferManager
        session={MOCK_SESSION_DATA}
        totalTransferReceived={5}
        totalTransferSent={5}
      />,
    );

    await waitFor(() => {
      expect(screen.getByText('Transfer Sent')).toBeInTheDocument();
    });

    const observerCallback = (global.IntersectionObserver as jest.Mock).mock
      .calls[0][0];

    observerCallback([{ isIntersecting: true }]);

    await waitFor(() => {
      expect(screen.getByText('Transfer Sent')).toBeInTheDocument();
    });

    expect(screen.getByText('Transfer Sent')).toBeInTheDocument();
  });

  test('should set hasMore to false when there are no more transactions', async () => {
    (getTransactionsByUserId as jest.Mock).mockResolvedValueOnce({
      ...MOCK_TRANSACTIONS_RESPONSE,
      meta: {
        pagination: {
          total: 5,
          page: 1,
          pageCount: 1,
        },
      },
    });

    render(
      <TransferManager
        session={MOCK_SESSION_DATA}
        totalTransferReceived={5}
        totalTransferSent={5}
      />,
    );

    await waitFor(() => {
      expect(screen.getByText('Transfer Sent')).toBeInTheDocument();
    });

    const observerCallback = (global.IntersectionObserver as jest.Mock).mock
      .calls[0][0];

    observerCallback([{ isIntersecting: true }]);

    await waitFor(() => {
      expect(
        screen.queryByText('Loading more transactions...'),
      ).not.toBeInTheDocument();
    });
  });

  test('should handle transfer type selection correctly', async () => {
    render(
      <TransferManager
        session={MOCK_SESSION_DATA}
        totalTransferReceived={5}
        totalTransferSent={5}
      />,
    );

    await waitFor(() => {
      expect(screen.getByText('Transfer Received')).toBeInTheDocument();
    });

    screen.getByText('Transfer Sent').click();

    await waitFor(() => {
      expect(getTransactionsByUserId).toHaveBeenCalledWith(
        MOCK_SESSION_DATA.user.id,
        expect.objectContaining({
          filters: { toAccountType: { $null: undefined } },
        }),
      );
    });

    screen.getByText('Transfer Received').click();

    await waitFor(() => {
      expect(getTransactionsByUserId).toHaveBeenCalledWith(
        MOCK_SESSION_DATA.user.id,
        expect.objectContaining({
          filters: { toAccountType: { $notNull: undefined } },
        }),
      );
    });
  });
});
