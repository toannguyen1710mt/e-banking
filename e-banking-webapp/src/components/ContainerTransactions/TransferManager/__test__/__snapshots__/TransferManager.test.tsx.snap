// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransferManager component should match snapshot 1`] = `
<div>
  <div
    class="mx-auto mb-[23px] flex flex-col justify-center gap-6 md:flex-row"
  >
    <div
      class="cursor-pointer"
    >
      <div
        class="flex flex-col relative overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none min-w-[180px] cursor-pointer p-4 sm:basis-1/2 bg-[#264653]"
        tabindex="-1"
      >
        <p
          class="text-foreground-100 text-base font-semibold text-white"
        >
          Transfer Received
        </p>
        <div
          class="flex items-center justify-between gap-4"
        >
          <span
            class="text-primary-200 text-[52px] font-bold text-white"
          >
            5
          </span>
          <div
            class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap h-6 text-tiny rounded-small px-2 py-1 text-primary bg-background-500/10"
          >
            <div
              class="max-h-[80%] text-white"
            >
              <svg
                class="w-[25px] h-4"
                fill="none"
                height="10"
                viewBox="0 0 11 10"
                width="11"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.66832 4.70416L5.51315 2.62083C5.47215 2.5829 5.42381 2.55316 5.3709 2.53333C5.26596 2.49166 5.14826 2.49166 5.04332 2.53333C4.99041 2.55316 4.94207 2.5829 4.90108 2.62083L2.7459 4.70416C2.70572 4.74301 2.67384 4.78913 2.65209 4.83989C2.63034 4.89065 2.61914 4.94506 2.61914 5C2.61914 5.11096 2.66474 5.21737 2.7459 5.29583C2.82707 5.37429 2.93715 5.41837 3.05194 5.41837C3.16672 5.41837 3.27681 5.37429 3.35797 5.29583L4.77608 3.92083V7.08333C4.77608 7.19384 4.82149 7.29982 4.90232 7.37796C4.98316 7.4561 5.09279 7.5 5.20711 7.5C5.32143 7.5 5.43106 7.4561 5.5119 7.37796C5.59273 7.29982 5.63815 7.19384 5.63815 7.08333V3.92083L7.05625 5.29583C7.09632 5.33488 7.14399 5.36588 7.19652 5.38704C7.24904 5.40819 7.30538 5.41908 7.36228 5.41908C7.41919 5.41908 7.47552 5.40819 7.52805 5.38704C7.58057 5.36588 7.62825 5.33488 7.66832 5.29583C7.70872 5.2571 7.74079 5.21101 7.76267 5.16024C7.78455 5.10946 7.79582 5.055 7.79582 5C7.79582 4.94499 7.78455 4.89053 7.76267 4.83976C7.74079 4.78898 7.70872 4.7429 7.66832 4.70416Z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <span
              class="flex-1 font-extrabold text-xs p-0 text-white"
            >
              8
              %
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="cursor-pointer"
    >
      <div
        class="flex flex-col relative overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none min-w-[180px] cursor-pointer p-4 sm:basis-1/2 bg-white"
        tabindex="-1"
      >
        <p
          class="text-foreground-100 text-base font-semibold text-primary-200"
        >
          Transfer Sent
        </p>
        <div
          class="flex items-center justify-between gap-4"
        >
          <span
            class="text-primary-200 text-[52px] font-bold text-primary-200"
          >
            5
          </span>
          <div
            class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap h-6 text-tiny rounded-small px-2 py-1 text-primary bg-primary-200 bg-opacity-10"
          >
            <div
              class="max-h-[80%] text-primary-200"
            >
              <svg
                class="w-[25px] h-4"
                fill="none"
                height="10"
                viewBox="0 0 11 10"
                width="11"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.66832 4.70416L5.51315 2.62083C5.47215 2.5829 5.42381 2.55316 5.3709 2.53333C5.26596 2.49166 5.14826 2.49166 5.04332 2.53333C4.99041 2.55316 4.94207 2.5829 4.90108 2.62083L2.7459 4.70416C2.70572 4.74301 2.67384 4.78913 2.65209 4.83989C2.63034 4.89065 2.61914 4.94506 2.61914 5C2.61914 5.11096 2.66474 5.21737 2.7459 5.29583C2.82707 5.37429 2.93715 5.41837 3.05194 5.41837C3.16672 5.41837 3.27681 5.37429 3.35797 5.29583L4.77608 3.92083V7.08333C4.77608 7.19384 4.82149 7.29982 4.90232 7.37796C4.98316 7.4561 5.09279 7.5 5.20711 7.5C5.32143 7.5 5.43106 7.4561 5.5119 7.37796C5.59273 7.29982 5.63815 7.19384 5.63815 7.08333V3.92083L7.05625 5.29583C7.09632 5.33488 7.14399 5.36588 7.19652 5.38704C7.24904 5.40819 7.30538 5.41908 7.36228 5.41908C7.41919 5.41908 7.47552 5.40819 7.52805 5.38704C7.58057 5.36588 7.62825 5.33488 7.66832 5.29583C7.70872 5.2571 7.74079 5.21101 7.76267 5.16024C7.78455 5.10946 7.79582 5.055 7.79582 5C7.79582 4.94499 7.78455 4.89053 7.76267 4.83976C7.74079 4.78898 7.70872 4.7429 7.66832 4.70416Z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <span
              class="flex-1 font-extrabold text-xs p-0 text-primary-200"
            >
              8
              %
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mb-4 flex flex-col gap-2"
  >
    <span
      class="text-primary-200 text-sm font-medium"
    >
      Transfer Request
       
      Received
    </span>
    <p
      class="text-primary-200 text-xs font-normal"
    >
      Manage your transfer by approving, declining request
    </p>
  </div>
  <div
    class="flex flex-col relative gap-4 w-full max-h-[672px] overflow-y-auto [&::-webkit-scrollbar]:w-0"
  >
    <table
      aria-describedby=""
      class="min-w-full h-auto table-auto w-full"
      id="react-aria-:r0:"
      role="grid"
      tabindex="0"
    >
      <thead
        class="sticky top-0 z-20 [&>tr]:first:shadow-small [&>tr]:first:rounded-none [&>tr:last-child]:hidden bg-background-900"
        role="rowgroup"
      >
        <tr
          class="group outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
          role="row"
        >
          <th
            class="group px-3 h-10 align-middle bg-default-100 whitespace-nowrap data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start text-xs opacity-50 last:rounded-none first:rounded-none text-primary-200 font-semibold"
            data-key="id"
            id="react-aria-:r0:-id"
            role="columnheader"
            tabindex="-1"
          >
            Transaction ID
          </th>
          <th
            class="group px-3 h-10 align-middle bg-default-100 whitespace-nowrap data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start text-xs opacity-50 last:rounded-none first:rounded-none text-primary-200 font-semibold"
            data-key="createdAt"
            id="react-aria-:r0:-createdAt"
            role="columnheader"
            tabindex="-1"
          >
            Transaction Date
          </th>
          <th
            class="group px-3 h-10 align-middle bg-default-100 whitespace-nowrap data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start text-xs opacity-50 last:rounded-none first:rounded-none text-primary-200 font-semibold"
            data-key="fromAccountType"
            id="react-aria-:r0:-fromAccountType"
            role="columnheader"
            tabindex="-1"
          >
            From
          </th>
          <th
            class="group px-3 h-10 align-middle bg-default-100 whitespace-nowrap data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start text-xs opacity-50 last:rounded-none first:rounded-none text-primary-200 font-semibold"
            data-key="amount"
            id="react-aria-:r0:-amount"
            role="columnheader"
            tabindex="-1"
          >
            Amount
          </th>
        </tr>
        <tr
          aria-hidden="true"
          class="w-px h-px block"
          style="margin-left: 0.25rem; margin-top: 0.25rem;"
          tabindex="-1"
        />
      </thead>
      <tbody
        class="divide-y divide-semiTransparentNavyBlue border-[0.2px] border-semiTransparentNavyBlue"
        data-empty="true"
        role="rowgroup"
      />
    </table>
  </div>
  <div
    class="mt-8 flex items-center justify-center text-center"
  >
    <div
      aria-label="Loading"
      class="relative inline-flex flex-col gap-2 items-center justify-center"
    >
      <div
        class="relative flex w-5 h-5"
      >
        <i
          class="absolute w-full h-full rounded-full animate-spinner-ease-spin border-solid border-t-transparent border-l-transparent border-r-transparent border-2 border-b-success"
        />
        <i
          class="absolute w-full h-full rounded-full opacity-75 animate-spinner-linear-spin border-dotted border-t-transparent border-l-transparent border-r-transparent border-2 border-b-success"
        />
      </div>
    </div>
  </div>
</div>
`;
